src/codeu/chat/client/core/UserContext.java

```diff
@@ -34,6 +40,68 @@ public UserContext(User user, BasicView view, BasicController controller) {
     this.view = view;
     this.controller = controller;
   }
+  
+  public User addUserInterest(String name) {
+	  Uuid idToSave = null;
+	  // find the user with the same name as the one given 
+	  for(User u: view.getUsers()) {
+		  if (name.equals(u.name)){
```

Comment:
oh shoot that's right. I will edit u-add to check for existing users with same name